frontend/src/components/common/ToastMessage.vue

```diff
@@ -0,0 +1,82 @@
+<script setup lang="ts">
+import axios from 'axios';
+import Toast from 'primevue/toast';
+import { useToast } from 'primevue/usetoast';
+import ErrorFilledIcon from '../icons/ErrorFilledIcon.vue';
+import { app } from '@/main';
+import {
+    useNotificationMessage,
+    useErrorDialog,
+} from '@/store/ApplicationState';
+
+const toast = useToast();
+
+const showToastTopRight = (sev: any, title: string, text: string) => {
+    toast.add({ severity: sev, summary: title, detail: text, group: 'tl' });
+};
+
+const onError = (error: any, info: string) => {
+    console.error(`Error occurred: ${error.toString()}`);
+    const genericErrorMsg = {
+        title: 'Error',
+        text: 'An application error has occurred. Please try again. If the error persists contact support.',
+    };
+
+    // Axios Http instance error that we like to pop out additional toast message.
+    if (axios.isAxiosError(error)) {
+        const err = error;
+        const axiosResponse = err.response;
+        const status = axiosResponse?.status;
+
+        const e401_authenticationErrorMsg = {
+            title: 'Error',
+            text: 'You are not logged in. Please log in.',
+        };
+
+        const e403_authorizationErrorMsg = {
+            title: 'Error',
+            text: 'You do not have the necessary authorization for the requested action.',
+        };
+
+        if (!status) {
+            showToastTopRight(
+                'error',
+                genericErrorMsg.title,
+                genericErrorMsg.text
+            );
+        } else if (status == 401) {
+            showToastTopRight(
+                'error',
+                e401_authenticationErrorMsg.title,
+                e401_authenticationErrorMsg.text
+            );
+        } else if (status == 403) {
+            showToastTopRight(
+                'error',
+                e403_authorizationErrorMsg.title,
+                e403_authorizationErrorMsg.text
+            );
+        } else if (status == 409) {
+            useNotificationMessage.isNotificationVisible = false;
+            useErrorDialog.isErrorVisible = true;
+        }
+        return;
+    }
+
+    showToastTopRight('error', genericErrorMsg.title, genericErrorMsg.text);
+};
+
+app.config.errorHandler = (err, instance, info) => {
+    onError(err, info);
+};
```

Comment:
Not really sure if messages will really show up using ToastMessage.vue....
Could you confirm those 4xx errors you coded above do work?
I kind of feel, maybe I am wrong, some configuration need to be registered as soon as possible in app before component is even initialized. Hope I am wrong, or this will break the behavior.